students/andykwon/lesson07/html_render.py

```diff
@@ -77,17 +92,64 @@ class Title(OneLineTag):
 
 class SelfClosingTag(Element):
 
-    def render(self, outfile)
-        pass
+    def append(self, *args, **kwargs):
+        raise TypeError
+
+    def open_tag(self,):
+
+        line = ''
+        opening = f"<{self.tag}"
+
+        if self.attr != {}:
+            for key, value in self.attr.items():
```

Comment:
this is ALMOST the same code as in the base class -- could you refactor so as not to repeat yourself?